Sources/Common/DataModel/Polygon/index.js

```diff
@@ -0,0 +1,248 @@
+import macro from 'vtk.js/Sources/macro';
+import * as vtkMath from 'vtk.js/Sources/Common/Core/Math';
+import vtkLine from 'vtk.js/Sources/Common/DataModel/Line';
+import vtkPlane from 'vtk.js/Sources/Common/DataModel/Plane';
+import vtkPriorityQueue from 'vtk.js/Sources/Common/Core/PriorityQueue';
+import { IntersectionState } from 'vtk.js/Sources/Common/DataModel/Line/Constants';
+
+// ----------------------------------------------------------------------------
+// vtkPolygon methods
+// ----------------------------------------------------------------------------
+
+const EPSILON = 1e-6;
+
+function vtkPolygon(publicAPI, model) {
+  // Set our classname
+  model.classHierarchy.push('vtkPolygon');
+
+  function computeNormal() {
+    const v1 = [0, 0, 0];
+    const v2 = [0, 0, 0];
+    model.normal = [0, 0, 0];
+    const anchor = [...model.firstPoint.point];
+
+    let point = model.firstPoint;
+    for (let i = 0; i < model.pointCount; i++) {
+      vtkMath.subtract(point.point, anchor, v1);
+      vtkMath.subtract(point.next.point, anchor, v2);
+
+      const n = [0, 0, 0];
+      vtkMath.cross(v1, v2, n);
```

Comment:
what happens if v1 is [0,0,0] ?

Reply:
If v1 is [0, 0, 0] then two of the three points in the triangle are the same. Since this triangle is empty it's normal shouldn't change the polygon's normal.